taipy/gui/gui.py

```diff
@@ -1143,6 +1143,41 @@ def __update_state_context(self, payload: dict):
             for var, val in state_context.items():
                 self._update_var(var, val, True, forward=False)
 
+
+    @staticmethod
+    def set_unsupported_data_converter(converter: t.Optional[t.Callable[[t.Any], t.Any]]) -> None:
+        """Set a custom converter for unsupported data types.
+
+        This function allows specifying a custom conversion function for data types that cannot
+        be serialized automatically. When Taipy GUI encounters an unsupported type in the data
+        being serialized, it will invoke the provided *converter* function. The returned value
+        from this function will then be serialized if possible.
+
+        Arguments:
+            converter: A function that converts a value with an unsupported data type (the only
+              parameter to the function) into data with a supported data type (the returned value
+              from the function).</br>
+              If set to `None`, it removes any existing converter.
+        """
+        Gui.__unsupported_data_converter = converter
+
+    @staticmethod
+    def _convert_unsupported_data(value: t.Any) -> t.Optional[t.Any]:
+        """
+        Handles unsupported data by invoking the converter if there is one.
+
+        Arguments:
+            value: The unsupported data encountered.
+
+        Returns:
+            The transformed data or None if no transformation is possible.
+        """
+        try:
+            return Gui.__unsupported_data_converter(value) if callable(Gui.__unsupported_data_converter) else None  # type: ignore
```

Comment:
use `_is_function` instead of `callable`

Reply:
I would recommend not to. 
A class (with `__call__()` is to be a valid value here.
Tell me.

Reply:
that's why `_is_function` was created

Reply:
but a class without `__call__` would be `callable`

Reply:
Whaat?